### PR TITLE
Fix Frame.abs to support bool type and disallow non-numeric types.

### PR DESCRIPTION
**Pull request by ueshin**
Fixes `Frame.abs` to support bool type and disallow non-numeric types.

### CODE REVIEW

File: databricks/koalas/tests/test_stats.py [hunk 1]
```diff
@@ -98,12 +98,22 @@ def test_abs(self):
...
                 "D": ["a", "b", "c", "d", "e"],
+                "E": [True, False, False, True, True],
             }
         )
         kdf = ks.from_pandas(pdf)
         self.assert_eq(kdf.A.abs(), pdf.A.abs())
         self.assert_eq(kdf.B.abs(), pdf.B.abs())
+        self.assert_eq(kdf.E.abs(), pdf.E.abs())
+        # pandas' bug?
+        # self.assert_eq(kdf[["B", "C", "E"]].abs(), pdf[["B", "C", "E"]].abs())
```
Comment (ueshin) [inline databricks/koalas/tests/test_stats.py:109]: The pandas result is:

```py
>>> pdf[["B", "C", "E"]].abs()
   B   C  E
0  1   6  1
1  2   7  0
2  3   8  0
3  4   9  1
4  5  10  1
```

whereas calculating separately:

```py
>>> pdf[["B", "C"]].abs()
     B     C
0  1.0   6.0
1  2.0   7.0
2  3.0   8.0
3  4.0   9.0
4  5.0  10.0
>>> pdf[["E"]].abs()
       E
0   True
1  False
2  False
3   True
4   True
```

I believe this is a pandas bug.

The current Koalas behavior:

```py
>>> ks.from_pandas(pdf)[["B", "C", "E"]].abs()
     B     C      E
0  1.0   6.0   True
1  2.0   7.0  False
2  3.0   8.0  False
3  4.0   9.0   True
4  5.0  10.0   True
```

File: databricks/koalas/generic.py [hunk 2]
```diff
@@ -1757,7 +1757,18 @@ def abs(self) -> Union["DataFrame", "Series"]:
...
-        return self._apply_series_op(lambda kser: kser._with_new_scol(F.abs(kser.spark.column)))
+
+        def abs(kser):
+            if isinstance(kser.spark.data_type, BooleanType):
+                return kser
+            elif isinstance(kser.spark.data_type, NumericType):
+                return kser.spark.transform(F.abs)
+            else:
+                raise TypeError(
+                    "bad operand type for abs(): {}".format(kser.spark.data_type.simpleString())
```
Comment (HyukjinKwon) [inline databricks/koalas/generic.py:1768]: Hm, should we show spark's dtype instead of Spark data type?

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

File: databricks/koalas/generic.py (hunk 2 above)
Comment (ueshin) [reply databricks/koalas/generic.py:1768]: In that case, we should address it separately. The same error message will appear in the other places.

File: databricks/koalas/tests/test_stats.py (hunk 1 above)
Comment (itholic) [reply databricks/koalas/tests/test_stats.py:109]: Seems pandas bug to me, too. Let me report this to pandas dev.

**Review by itholic — APPROVED** [review]
LGTM. Let's address https://github.com/databricks/koalas/pull/1980#discussion_r547607957 in separately.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**